### PR TITLE
Feature/978 Bower Support for Plugins

### LINKED ISSUE #978: Add plugin bower dependency support
It would be nice to declare bowerDependencies for a plugin in the details.json

### PR DESCRIPTION
**Pull request by brianhyder**
Fixes #978 

- [x] Unit tests created and pass
- [x] JS Docs have been updated

**Description:**
Provides the ability for plugins to specify bower dependencies.  A HUGE thanks to @n1ru4l for the inspiration and leg work.

@pencilblue/owners

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,74 +1,75 @@
...
-    "bugs": {
-        "url": "https://github.com/pencilblue/pencilblue/issues"
-    },
-    "_id": "pencilblue@0.6.0"
+    {
+      "name": "Brian Hyder"
+    }
+  ],
+  "name": "pencilblue",
+  "description": "A full featured Node.js CMS and blogging platform (plugins, server cluster management, data-driven pages)",
```
Comment (blakecallens) [inline package.json]: Let's add our new description from the github page:
```
Business class content management for Node.js (plugins, server cluster management, data-driven pages)
```

File: package.json [hunk 2]
```diff
@@ -1,74 +1,75 @@
...
+  "engines": {
+    "node": ">= 0.11.0 <=6"
+  },
+  "scripts": {
+    "start": "node pencilblue.js",
+    "test": "istanbul cover ./node_modules/mocha/bin/_mocha -- -R spec --recursive"
+  },
+  "licenses": [
+    {
+      "type": "GNU GENERAL PUBLIC LICENSE Version 3",
```
Comment (blakecallens) [inline package.json]: I think the official syntax for this is ```GPL-3.0```

**Comment by blakecallens** [comment]
Just a couple comments on the package.json while we're updating it. Programmatic changes look good to me.

**Comment by brianhyder** [comment]
Awesome. Thank you, sir. I'll make them tomorrow and merge.

**Pull request merged**